apps/server/src/shared/domain/rules/lesson.rule.ts

```diff
@@ -26,9 +27,83 @@ export class LessonRule extends BasePermission<Lesson> {
 		}
 
 		const hasPermission = this.utils.hasAllPermissions(user, requiredPermissions);
-		const hasCoursePermission = this.courseRule.hasPermission(user, entity.course, { action, requiredPermissions: [] });
+		let hasCoursePermission = false;
+		if (entity.course) {
+			hasCoursePermission = this.courseRule.hasPermission(user, entity.course, { action, requiredPermissions: [] });
+		} else if (entity.courseGroup) {
+			hasCoursePermission = this.courseGroupRule.hasPermission(user, entity.courseGroup, {
+				action,
+				requiredPermissions: [],
+			});
+		}
 		const result = hasPermission && hasCoursePermission;
 
+		return result;
+	} */
+
+	public hasPermission(user: User, entity: Lesson, context: IPermissionContext): boolean {
```

Comment:
A conter part to my comment in coursegroup rule what is when i enter this code with context.action = undefined? I do not think about by the draft. Can we run into this, or do we have always action = 'read' if nothing is passed?

```
export interface IPermissionContext {
	action?: Actions;
	requiredPermissions: Permission[];
}
```

I will check the permissionContextBuilder and the code if it is used without. But the interface has no default value and this is crititcal. Also critical is that we have no test for it that failed or say what should going on when action is empty.